src/evaluator/functions.rs

```diff
@@ -870,6 +871,105 @@ pub fn fn_random<'a>(
     Ok(Value::number(context.arena, v))
 }
 
+pub fn fn_now<'a>(
+    context: FunctionContext<'a, '_>,
+    args: &[&'a Value<'a>],
+) -> Result<&'a Value<'a>> {
+    let now = Utc::now();
+
+    // Use pattern matching to extract picture and timezone
+    let (picture, timezone) = match args {
+        [picture, timezone] => (picture.as_str(), timezone.as_str()),
+        [picture] => (picture.as_str(), Cow::Borrowed("")),
+        [] => (Cow::Borrowed(""), Cow::Borrowed("")),
+        &[..] => return Ok(Value::string(context.arena, "")), // Handle case with more than two arguments
+    };
+
+    // Handle the default case (no picture, no timezone) -> return ISO 8601 timestamp in UTC
+    if picture.is_empty() && timezone.is_empty() {
+        return Ok(Value::string(
+            context.arena,
+            &now.to_rfc3339_opts(chrono::SecondsFormat::Millis, true),
+        ));
+    }
+
+    // Adjust the time for the given timezone if provided
+    let adjusted_time = if !timezone.is_empty() {
+        parse_timezone_offset(&timezone)
+            .map(|offset| now.with_timezone(&offset))
+            .unwrap_or(now.into()) // Fallback to UTC if invalid timezone
+    } else {
+        now.into() // Use UTC if no timezone provided
+    };
+
+    // If a valid picture is provided, format the time accordingly
+    if !picture.is_empty() {
+        return Ok(Value::string(
+            context.arena,
+            &format_custom_date(&adjusted_time, &picture),
+        ));
+    }
+
+    // Return an empty string if the picture is empty but a valid timezone is provided
+    Ok(Value::string(context.arena, ""))
+}
+
+// Custom formatting function based on the 'picture' string
+fn format_custom_date(date: &DateTime<FixedOffset>, picture: &str) -> String {
+    match picture {
```

Comment:
> A potential middle ground between this code and full support though could be splitting up the picture string into the variable components and literal components, eg [Variable("H01"), Literal(":"), Variable("m01")], etc. Then, have a match on the variable components (only support a subset for now), but allow the supported variable strings to be composed in any combination.

Handled this into a new reusable fn `format_custom_date`

> We should also structure this code so that picture string handling is shared with the (currently unimplemented) $fromMillis function, so that we're able to rely on the comprehensive test cases in tests/testsuite/skip/function-fromMillis/formatDateTime.json.

This i can take up as next fn change.